3IPs/3ip-10.md

```diff
@@ -0,0 +1,299 @@
+---
+3ip: 10
+title: 3ID JSON-RPC
+author: oed@3box.io
+discussions-to: https://chat.3box.io
+status: Draft
+created: 2019-09-02
+requires: 7
+---
+
+## Simple Summary
+This 3IP describes the json-rpc interface which wallets can implement to natively support 3Box.
+
+## Abstract
+The json-rpc described in this document outlines the interface for all functions that are needed by the 3box-js client when interacting with an identity. The IdentityWallet SDK will natively support these functions, making it easy for wallets to add 3id support.
+
+## Motivation
+By specifying a json-rpc interface we make the 3ID system more modular. We can allow wallets to natively support 3box through an injected provider, or systems like WalletConnect. It also makes it simpler for non ethereum wallets to support 3box as well.
+
+## Specification
+
+### `3id_authenticate`
+
+Method for authenticating to get access to the requested spaces.
+
+#### **Parameters:**
+
+1. `spaces` - An array of space names as String
+2. `authData` - An array of authData objects
+3. `address` - String, The address associated with this 3ID (optional)
+
+#### **Returns:**
+
+An object containing the public keys for the main account and for any requested space.
+
+#### **Example:**
+
+```json
+    // Request
+    {
+      "id": 1,
+      "jsonrpc": "2.0",
+      "method": "3id_authenticate",
+      "params": {
+        "spaces": ["space1", "space2"],
+        "authData": []
+      }
+    }
+
+    // Result
+    {
+      "id": 1,
+      "jsonrpc": "2.0",
+      "result": {
+        "main": {
+          "signingkey": "0xexample",
+          "managementkey": "0xexample",
+          "encryptionkey": "0xexample"
+        },
+        "spaces": {
+          "space1": {
+            "signingKey": "0xexample",
+            "encryptionKey": "0xexample"
+          },
+          "space2": {
+            "signingKey": "0xexample",
+            "encryptionKey": "0xexample"
+          }
+        }
+      }
+    }
+```
+
+### `3id_isAuthenticated`
```

Comment:
is this stateful, or is just answering the question, is there any authData for this here?

Reply:
It's stateful yes. It returns true if the app making the call is authenticated to access the given spaces.